src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,309 @@
+import org.apache.catalina.*;
+import org.apache.catalina.session.ManagerBase;
+import org.apache.commons.lang3.time.DateFormatUtils;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import org.apache.commons.lang3.SerializationUtils;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.*;
+
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import redis.clients.jedis.Transaction;
+
+/**
+ * Created by Cesar Valverde on 6/10/2015.
+ */
+public class RedisSessionManagerT8 extends ManagerBase{
+
+    private static final Log log = LogFactory.getLog(RedisSessionManagerT8.class);
+
+    private static final String REDIS_METADATA_KEY = ":metadata";
+    private static final String REDIS_ATTRIBUTES_KEY = ":attributes";
+
+    //Metadata keys
+    public static final String METADATA_VALID = "valid";
+    public static final String METADATA_CREATION_TIME = "creation_time";
+    public static final String METADATA_LAST_ACCESS_TIME = "last_access_time";
+    public static final String METADATA_MAX_INACTIVE_INTERVAL = "max_inactive_interval";
+
+    protected JedisPool redisConnectionPool = null;
+    protected JedisPoolConfig redisConnectionPoolConfig = null;
+
+    //Database info
+    private String dbHost = "localhost";
+    private int dbPort = 2222;
+    private int dbTimeout = 0;
+    private String dbPassword = "";
+
+    @Override
+    public void load() throws ClassNotFoundException, IOException {
+    }
+
+    @Override
+    public void unload() throws IOException {
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START Sessions
+    //------------------------------------------------------------------------------------------------------------------
+
+    @Override
+    public int getRejectedSessions() {
+        // Essentially do nothing.
+        return 0;
+    }
+
+    @Override
+    protected RedisSessionT8 getNewSession() {
+        return new RedisSessionT8(this);
+    }
+
+    @Override
+    public Session createSession(String requestedSessionId) {
+
+        RedisSessionT8 session = (RedisSessionT8) createEmptySession();
```

Comment:
Adding a cast should be physically painful to you. If this class will always produce a `RedisSessionT8`, then override `createEmptySession()` and put this code into there:

```java
@Override
protected RedisSessionT8 createEmptySession() {
  return (RedisSessionT8)super.createEmptySession();
}
```